File: discover_ruby/manipulate_the_data.rb
require 'HTTPClient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 170f8634262556331b3cb1809ffd4eb6e6fd40e1'
}

clnt = HTTPClient.new
response = clnt.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', nil, extheaders) #replaced 'query = nil' by 'nil'
json = JSON.parse(response) #parsing JSON string
items = json["items"] #converting JSON string into Ruby hash

full_names = items.map{ |item|  item["full_name"] }.join("\n") #creates hash with key items using Array#map and Array#join methods

# Array#map returns a new array filled with whatever gets returned by the block each time it runs
# Array#join method (the argument to join is what to insert between the strings - in this case a new line):

puts full_names #prints the full names of the most popular Ruby projects on GitHub

File: discover_more_python/manipulate_the_data.py
""" This program parses the response from Github's API when requesting 
    the 30 most popular Python projects to a file into a JSON format and
    prints the name of each project
"""

from urllib2 import Request, urlopen, URLError
import json

request_headers = {
    'User-Agent': 'Holberton School',
    'Authorization': 'token 170f8634262556331b3cb1809ffd4eb6e6fd40e1'
}

request = Request('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', headers=request_headers)

try:
            response = urlopen(request)
            string = response.read()
            parsed = json.loads(string)

            for item in parsed["items"]:
                        print item.get("full_name")
except URLError, e:
        print "Error code: ", e
        
